nova_cli/src/main.rs

```diff
@@ -158,72 +165,53 @@ fn main() -> Result<(), Box<dyn std::error::Error>> {
                 }
             }
         }
-    }
-
-    Ok(())
-}
-
-fn exit_with_parse_errors(errors: Vec<OxcDiagnostic>, source_path: &str, source: &str) -> ! {
-    assert!(!errors.is_empty());
-
-    // This seems to be needed for color and Unicode output.
-    miette::set_hook(Box::new(|_| {
-        use std::io::IsTerminal;
-        let mut handler = oxc_diagnostics::GraphicalReportHandler::new();
-        if !std::io::stdout().is_terminal() || std::io::stderr().is_terminal() {
-            // Fix for https://github.com/oxc-project/oxc/issues/3539
-            handler = handler.with_theme(GraphicalTheme::none());
-        }
-        Box::new(handler)
-    }))
-    .unwrap();
-
-    eprintln!("Parse errors:");
-
-    // SAFETY: This function never returns, so `source`'s lifetime must last for
-    // the duration of the program.
-    let source: &'static str = unsafe { std::mem::transmute(source) };
-    let named_source = miette::NamedSource::new(source_path, source);
-
-    for error in errors {
-        let report = error.with_source_code(named_source.clone());
-        eprint!("{:?}", report);
-    }
-    eprintln!();
-
-    std::process::exit(1);
-}
+        Command::Repl { verbose } => {
+            let allocator = Default::default();
+            let host_hooks: &CliHostHooks = &*Box::leak(Box::default());
+            let mut agent = Agent::new(
+                Options {
+                    disable_gc: false,
+                    print_internals: verbose,
+                },
+                host_hooks,
+            );
+            {
+                let create_global_object: Option<fn(&mut Realm) -> Object> = None;
+                let create_global_this_value: Option<fn(&mut Realm) -> Object> = None;
+                initialize_host_defined_realm(
+                    &mut agent,
+                    create_global_object,
+                    create_global_this_value,
+                    Some(initialize_global_object),
+                );
+            }
+            let realm = agent.current_realm_id();
 
-fn initialize_global_object(agent: &mut Agent, global: Object) {
-    use nova_vm::ecmascript::{
-        builtins::{create_builtin_function, ArgumentsList, Behaviour, BuiltinFunctionArgs},
-        execution::JsResult,
-        types::{InternalMethods, IntoValue, PropertyDescriptor, PropertyKey},
-    };
-
-    // `print` function
-    fn print(agent: &mut Agent, _this: Value, args: ArgumentsList) -> JsResult<Value> {
-        if args.len() == 0 {
-            println!();
-        } else {
-            println!("{}", args[0].to_string(agent)?.as_str(agent));
+            loop {
+                let mut input = String::new();
+                std::io::stdin().read_line(&mut input)?;
```

Comment:
Lines should have a prompt, right? Maybe:
```suggestion
                print!("> ");
                std::io::stdin().read_line(&mut input)?;
```

Reply:
I was testing that and it was double printing after errors so I'll look into that after